Helper/Data.php

```diff
@@ -93,31 +95,14 @@ public function getGeneralConfig(string $code = '', int $storeId = null)
         return $this->getConfigValue(self::XML_PATH_GENERAL . $code, $storeId);
     }
 
-    /**
-     * @return mixed
-     * @throws \Exception
-     */
-    public function getAccountSettings()
-    {
-        $accountSettings = $this->getGeneralConfig('account_settings');
-        if (! $accountSettings) {
-            throw new \RuntimeException(
-                'No account settings found. Press the import button in general configuration to fetch account settings.'
-            );
-        }
-
-        return unserialize($accountSettings);
-    }
-
     /**
      * @throws \Exception
      */
-    public function getDropOffPoint(string $carrier): DropOffPoint
+    public function getDropOffPoint(AbstractCarrier $carrier): DropOffPoint
     {
-        $accountSettings                 = $this->getAccountSettings();
-        $carrierConfigurationsCollection = $accountSettings->all()['carrier_configurations']->all();
-        $carrierConfiguration            = $carrierConfigurationsCollection[CarrierPostNL::NAME === $carrier ? 0 : 1];
-        $dropOffPoint                    = $carrierConfiguration->getDefaultDropOffPoint();
+        $accountSettings = AccountSettings::getInstance();
+        $carrierConfiguration = $accountSettings->getCarrierConfigurationByCarrierId($carrier->getId());
```

Comment:
na wat we besproken hebben misschien wel mooier om hier getCarrierConfigurationByCarrier van te maken en de hele carrier erin te gooien